src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayMiscellaneousOptions.cs

```diff
@@ -72,5 +72,20 @@ public enum SymbolDisplayMiscellaneousOptions
         /// Append '!' to non-nullable reference types.
         /// </summary>
         IncludeNotNullableReferenceTypeModifier = 1 << 8,
+
+        /// <summary>
+        /// Includes <c>implicit</c> or <c>explicit</c> keywords for user-defined operators.
+        /// </summary>
+        /// <remarks>
+        /// This is useful for explicit interface members completion where we expect something like:
+        /// <code><![CDATA[
+        /// static implicit I2<Test2>.operator int(Test2 x)
+        /// ]]></code>
+        /// instead of:
+        /// <code><![CDATA[
+        /// static implicit I2<Test2>.implicit operator int(Test2 x)
```

Comment:
I don't see why we would ever format this text. That just feels like a bug.

Reply:
i am a little confused how symboldisplay plays into this scenario.  i would *not* expect completion to be using symboldisplay here.  that feels like a core bug on its own to me.

Reply:
That too.

Reply:
https://github.com/dotnet/roslyn/blob/bcd1dab54f60e40940da2aa909c8542f31a6d4b4/src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.cs#L101

:(

@CyrusNajmabadi @333fred Any suggestion on how to best approach this?

Reply:
1. kill me
2. replace that code with code that just builds the right syntax for what we're generating.  it seems not that hard right?  it's just member name, type parameters, parameters, constraints.

Reply:
@CyrusNajmabadi 

> replace that code with code that just builds the right syntax for what we're generating

I did that. The code isn't large as I was initially expecting. The only risk I see with this approach is regressing some edge cases, but I'll add more tests.